src/input/playlist.rs

```diff
@@ -99,6 +104,31 @@ impl CurrentProgram {
                     self.json_mod = json.modified;
                     *self.nodes.lock().unwrap() = json.program;
 
+                    self.get_current_clip();
+                    self.index.fetch_add(1, Ordering::SeqCst);
+                }
+            }
+        } else if is_remote(&self.json_path.clone().unwrap()) {
```

Comment:
I think this can be also merged with the existing condition. For example:

```Rust
} else if Path::new(&self.json_path.clone().unwrap()).is_file()
    || is_remote(...) {
    let mod_time = modified_time(&self.json_path.clone().unwrap());  // this function can be extend to check mod time from remote sources

    if self.json_mod != mod_time.map(|m| m.to_string()) {
        ...
```